fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/ListKeyExpression.java

```diff
@@ -148,6 +152,16 @@ public int getColumnSize() {
         return RecordMetaDataProto.KeyExpression.newBuilder().setList(toProto()).build();
     }
 
+    @Nonnull
+    @Override
+    public KeyExpression normalizeForPlanner(@Nonnull Source rootSource, @Nonnull Function<Element, Element> elementModifier) {
```

Comment:
Hmmm, I'm not sure. Should this return a single....`TupleElement` and then that can be a source to another set of elements to extract elements from the nested tuple? Or something like that? Because even though opening it up and adding predicates on the nested elements isn't quite something we need yet, it probably is something we'll want eventually.

You can't quite flatten it because the sorting behavior is subtly different.

Reply:
I think that we've identified that `ListKeyExpression` is wonky from a planning perspective. I've made a note to look into that in the future.